collector/thermalzone.go

```diff
@@ -0,0 +1,105 @@
+package collector
+
+import (
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+func init() {
+	Factories["thermalzone"] = NewThermalZoneCollector
+}
+
+// A ThermalZoneCollector is a Prometheus collector for WMI Win32_PerfRawData_Counters_ThermalZoneInformation metrics
+type ThermalZoneCollector struct {
+	HighPrecisionTemperature *prometheus.Desc
+	PercentPassiveLimit      *prometheus.Desc
+	Temperature              *prometheus.Desc
+	ThrottleReasons          *prometheus.Desc
+}
+
+// NewThermalZoneCollector ...
+func NewThermalZoneCollector() (Collector, error) {
+	const subsystem = "thermalzone"
+	return &ThermalZoneCollector{
+		HighPrecisionTemperature: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "high_precision_temperature"),
```

Comment:
We should suffix this with the unit, so `_kelvin` in this case (unless we convert to Celsius?).
On closer inspection, the value looks like it is actually "decikelvin"(?), I have a value of 3010

Reply:
I'd be more than happy to convert to Celsius, fits our use case better anyway. Wasn't sure if y'all modified the generated code, or if you just wanted a 1 to 1 mapping to the powershell commands.

Reply:
The generated code is just to get a starting point and not have to write so much boilerplate by hand all the time :)